src/Tizen.NUI/src/public/Images/NativeImageSource.cs

```diff
@@ -38,18 +38,14 @@ internal NativeImageSource(IntPtr cPtr, bool cMemoryOwn) : base(Interop.NativeIm
         }
 
         /// <summary>
-        /// Get URI from native image source.
+        /// Generate Url from current native image source.
         /// </summary>
         [EditorBrowsable(EditorBrowsableState.Never)]
-        public string Url
+        public ImageUrl GenerateCurrentUrl()
```

Comment:
You do better keep API, and manage a lifecycle of ImageUrl in `NativeImageSource`


``` c#
ImageUrl _imageUrl = null;
public string Url 
{
    get 
    {
          _imageUrl?.Dispose();
          _imageUrl = new ImageUrl(Interop.NativeImageSource.GenerateUrl(this.SwigCPtr.Handle), true);
            if (NDalicPINVOKE.SWIGPendingException.Pending) throw new InvalidOperationException("FATAL: get Exception", NDalicPINVOKE.SWIGPendingException.Retrieve());
         return _imageUrl.ToString();
    }
}

```

Reply:
GenerateCurrentUrl() changes every time it's called,
so I think the method is better than the property.
What do you think?

Reply:
Even though every times called, the content of `ImageUrl` will not changed if `NativeImageSource` was not changed. 
So, I thinks it is a property of `NativeImageSource`

Reply:
If I optimize it, I will add a boolean field to check buffer was dirty and if image buffer was dirty, I will generate a new ImageUrl, if not I will return cached ImageUrl, 


``` c#

bool _isDirty = false;
public IntPtr AcquireBuffer(ref int width, ref int height, ref int stride) 
{
   _isDirty = true;
 ...
}


ImageUrl _imageUrl = null;
public string Url 
{
    get 
    {
          if (!_isDirty && _imageUrl != null)
                return _imageUrl;

          _imageUrl?.Dispose();
           _isDirty = false;
          _imageUrl = new ImageUrl(Interop.NativeImageSource.GenerateUrl(this.SwigCPtr.Handle), true);
            if (NDalicPINVOKE.SWIGPendingException.Pending) throw new InvalidOperationException("FATAL: get Exception", NDalicPINVOKE.SWIGPendingException.Retrieve());
         return _imageUrl.ToString();
    }
}
```

Reply:
Unfortunately, ImageUrl seems to be newly created every time even if nativeImageSource is not changed.
For example, texture needs to be created internally when the raw buffer is changed.
so I think the method is better.

Reply:
If you want to provide as method, I recommend to rename like `GenerateSnapshot`
I don't know the detail implemention of it, but `ImageUrl` seem to not connect a `NativeImageSource`, it looks just a snapshot if `NativeImageSource`, 
So in this view, Method will be better than property

Reply:
because of this problem, I added Get"Current" Url but it still ambiguous.
I will think more about this.